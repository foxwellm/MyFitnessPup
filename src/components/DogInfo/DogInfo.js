import React from 'react'

export const DogInfo = ({ dogFromStore: { id, name, age, gender, contact, description, url, status, distance, photo } }) => {
  const dogName = !name || !name.match(' ') ? name : name.split(' ')[0]
  return (
    <div className='DogInfo-container'>
      <div className='doginfo-img-container'>
        <img className='doginfo-img' alt='dog from shelter' src={photo}></img>
      </div>
      <div className='doginfo-info-container'>
        <div className='doginfo-name'>{dogName}</div>
        <div className='doginfo doginfo-age'>Age: {age}</div>
        <div className='doginfo doginfo-gender'>Gender: {gender}</div>
        <div className='doginfo doginfo-distance'>Location: {distance} away</div>
        <div className='doginfo doginfo-status'>Status: {status}</div>
        <div className='doginfo doginfo-contact'>Contact: {contact}</div>
        <div className='doginfo doginfo-url'>PetFinder URL: {url}</div>
        <div className='doginfo doginfo-description'>Description: {description}</div>
      </div>
    </div>
  )
}

export default DogInfo