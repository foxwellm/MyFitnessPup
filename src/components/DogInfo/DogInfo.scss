.DogInfo-container {
  display: flex;
  width: 90vw;
  height: 40vw;
  margin: 50px auto;
  text-align: left;
}

.doginfo-img-container {
  min-width: 300px;
  min-height: 450px;
  width: 30vw;
  height: 40vw;
}

.doginfo-info-container {
  width: 55vw;
  margin: 2vw 0 2vw 5vw;
}

.doginfo-img {
  width: 100%;
  height: 100%;
}

.doginfo {
  font-size: 1.5vw;
  padding: 0.2vw;
}

.doginfo-name {
  font-size: 5vw;
}
