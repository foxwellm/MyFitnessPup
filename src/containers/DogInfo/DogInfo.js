import React, {Component} from 'react'

export class DogInfo extends Component {
  render() {
    return (
      <div>
        Dog Info
      </div>
    )
  }
}

export default DogInfo