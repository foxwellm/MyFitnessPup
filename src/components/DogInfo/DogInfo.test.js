import React from 'react'
import { shallow } from 'enzyme'
import DogInfo from './DogInfo'

describe('DogInfo', () => {
  let wrapper
  let dogFromStore

  beforeEach(() => {
    dogFromStore = {
      id: 6,
      name: 'Butch',
      age: 'Young',
      gender: 'male',
      contact: 'contact@gmail.com',
      desription: 'description',
      url: 'petfinder.com/dog123',
      status: 'available',
      distance: '12 mi',
      photo: 'petfinder/photoUrl'
    }
    wrapper = shallow(<DogInfo
      dogFromStore={dogFromStore}
    />)
  })
  it('should have match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have match the correct snapshot when the name is more than one word', () => {
    dogFromStore.name = 'Butch Cassidy'
    wrapper.setProps({dogFromStore})
    expect(wrapper).toMatchSnapshot()
  })
})