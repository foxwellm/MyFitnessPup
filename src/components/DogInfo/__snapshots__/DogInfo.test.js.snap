// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogInfo should have match the correct snapshot 1`] = `
<div
  className="DogInfo-container"
>
  <div
    className="doginfo-img-container"
  >
    <img
      alt="dog from shelter"
      className="doginfo-img"
      src="petfinder/photoUrl"
    />
  </div>
  <div
    className="doginfo-info-container"
  >
    <div
      className="doginfo-name"
    >
      Butch
    </div>
    <div
      className="doginfo doginfo-age"
    >
      Age: 
      Young
    </div>
    <div
      className="doginfo doginfo-gender"
    >
      Gender: 
      male
    </div>
    <div
      className="doginfo doginfo-distance"
    >
      Location: 
      12 mi
       away
    </div>
    <div
      className="doginfo doginfo-status"
    >
      Status: 
      available
    </div>
    <div
      className="doginfo doginfo-contact"
    >
      Contact: 
      contact@gmail.com
    </div>
    <div
      className="doginfo doginfo-url"
    >
      PetFinder URL: 
      petfinder.com/dog123
    </div>
    <div
      className="doginfo doginfo-description"
    >
      Description: 
    </div>
  </div>
</div>
`;

exports[`DogInfo should have match the correct snapshot when the name is more than one word 1`] = `
<div
  className="DogInfo-container"
>
  <div
    className="doginfo-img-container"
  >
    <img
      alt="dog from shelter"
      className="doginfo-img"
      src="petfinder/photoUrl"
    />
  </div>
  <div
    className="doginfo-info-container"
  >
    <div
      className="doginfo-name"
    >
      Butch
    </div>
    <div
      className="doginfo doginfo-age"
    >
      Age: 
      Young
    </div>
    <div
      className="doginfo doginfo-gender"
    >
      Gender: 
      male
    </div>
    <div
      className="doginfo doginfo-distance"
    >
      Location: 
      12 mi
       away
    </div>
    <div
      className="doginfo doginfo-status"
    >
      Status: 
      available
    </div>
    <div
      className="doginfo doginfo-contact"
    >
      Contact: 
      contact@gmail.com
    </div>
    <div
      className="doginfo doginfo-url"
    >
      PetFinder URL: 
      petfinder.com/dog123
    </div>
    <div
      className="doginfo doginfo-description"
    >
      Description: 
    </div>
  </div>
</div>
`;
